packages/cdk/lib/artifacts/batch-artifacts/ecs-additions/provision.sh

```diff
@@ -45,39 +60,67 @@ function ecs() {
     fi
 }
 
-# make sure that docker and ecs are running on script exit to avoid
+function getArtifactRoot(){
+  url=$(\
+      aws ssm get-parameter \
+          --name /"${ARTIFACTS_NAMESPACE}"/_common/installed-artifacts/s3-root-url \
+          --query 'Parameter.Value' \
+          --output text \
+  )
+  echo "$url"
+}
+
+function errorOrInt(){
+  echo "WARNING - received a $1 signal"
+
+  TOKEN=$(curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+  INSTANCE_ID=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" -s "http://169.254.169.254/latest/meta-data/instance-id")
+
+  echo " Attempting to mark myself, $INSTANCE_ID, unhealthy so the autoscaler can start a replacement"
+  aws autoscaling set-instance-health --instance-id "$INSTANCE_ID"  --health-status Unhealthy || echo "ERROR could not mark $INSTANCE_ID unhealthy, shutting down" && shutdown now
+}
+
+# make sure that docker and ecs are running on script exit or clean up on failure to avoid
 # zombie instances
-trap "ecs enable" INT ERR EXIT
+trap "ecs enable" EXIT
+trap "errorOrInt INT" INT
+trap "errorOrInt ERR" ERR
 
 set +e
 ecs disable
 set -e
 
-ARTIFACT_S3_ROOT_URL=$(\
-    aws ssm get-parameter \
-        --name /${ARTIFACTS_NAMESPACE}/_common/installed-artifacts/s3-root-url \
-        --query 'Parameter.Value' \
-        --output text \
-)
+ARTIFACT_S3_ROOT_URL=$(getArtifactRoot)
+if [[ -v ${ARTIFACT_S3_ROOT_URL} ]]; then
+    echo "ARTIFACT_S3_ROOT_URL not found, trying again" && sleep 5 && ARTIFACT_S3_ROOT_URL=$(getArtifactRoot)
+fi
+echo "ARTIFACT_S3_ROOT_URL = $ARTIFACT_S3_ROOT_URL"
 
 
 # retrieve and install amazon-ebs-autoscale
+echo "WORKFLOW_ORCHESTRATOR = $WORKFLOW_ORCHESTRATOR"
 if [ "$WORKFLOW_ORCHESTRATOR" != "miniwdl" ]; then
+  echo "obtaining amazon-ebs-autoscale artifacts"
   cd /opt
-  sh $BASEDIR/get-amazon-ebs-autoscale.sh \
+  sh "$BASEDIR"/get-amazon-ebs-autoscale.sh \
       --install-version dist_release \
       --artifact-root-url "$ARTIFACT_S3_ROOT_URL" \
       --file-system btrfs \
       --initial-size "$INITIAL_EBS_SIZE"
+  echo "amazon-ebs-autoscale artifacts installed with return code = $?"
```

Comment:
I believe these "echo error codes" would always print 0, because we have the "set -e" above which would exit the script immediately if any command returns non-zero

Reply:
They will right now. Main purpose of the echo is to log that the sub process is done. If we ever remove the `-e` it will be useful to know if the sub process errored